src/main/java/com/example/onlineservicesemulator/Controller.java

```diff
@@ -49,4 +92,61 @@ public double calculateLongestServiceString(List<String> servicesNames) {
         }
         return max;
     }
+
+    public void addFiles(String serviceName) {
+        addFileButton.setOnMouseClicked(mouseEvent -> {
+            try {
+                FXMLLoader loader = new FXMLLoader(getClass().getResource("uploadFilesWindow.fxml"));
+                Parent root = loader.load();
+
+                UploadFilesWindowController uploadFilesController = loader.getController();
+                uploadFilesController.setServiceName(serviceName);
+                uploadFilesController.setMainController(this);
+                Stage stage = new Stage();
+                stage.setScene(new Scene(root));
+
+                stage.show();
+            } catch (Exception exception) {
+                exception.printStackTrace();
+            }
+        });
+    }
+
+    public void setRemoveFileButtonListener() {
+        removeFileButton.setOnMouseClicked(new EventHandler<MouseEvent>() {
+            @Override
+            public void handle(MouseEvent event) {
+                String fileToBeRemoved = filesList.getSelectionModel().getSelectedItem();
+                if (!fileToBeRemoved.isEmpty()) {
+                    filesList.getItems().remove(fileToBeRemoved);
+                    servicesAndUploadedFilesMap.remove(fileToBeRemoved);
+                    removeFileFromDisk(fileToBeRemoved);
+                }
+            }
+        });
+    }
+
+    private void removeFileFromDisk(String fileToBeRemoved) {
+        Path uploadedFilesDir = Paths.get(".\\uploadedfiles");
+        Path filePath = uploadedFilesDir.resolve(fileToBeRemoved);
+        if (Files.exists(filePath)) {
+            try {
+                Files.delete(filePath);
+                System.out.println("File deleted successfully: " + filePath);
```

Comment:
remember that system outs are only useful when you run the application in an idea or cmd, but what if you're running the application as a standalone app?